pytorch_lightning/plugins/training_type/training_type_plugin.py

```diff
@@ -49,9 +49,8 @@ def __init__(
     ) -> None:
         self._accelerator = accelerator
         self._model: Optional[Module] = None
-        checkpoint_io = checkpoint_io if checkpoint_io is not None else TorchCheckpointIO()
-        self._checkpoint_io = checkpoint_io
-        self._precision_plugin = precision_plugin
+        self.checkpoint_io = checkpoint_io
+        self.precision_plugin = precision_plugin
```

Comment:
```suggestion
        self._checkpoint_io = checkpoint_io
        self._precision_plugin = precision_plugin
```

Reply:
Can you explain why? Don't we want the subclasses that override the `setter` with validation (e.g. tpu) to run the validation on init?

See the test I just added in my last commit. It would fail with your proposed change

Reply:
Yeah sure. 
To distinguish user passed Trainer(strategy = CustomStrategy()) Versus. Trainer(stategy = CustomStrategy(precision_plugin = PrecisionPlugin())) in accelerator_connector
If we call strategy.precision_plugin, it will execute the @property which will assign a default value. Then we don't know did user pass in None or a precision_plugin. The current accelerator logic is depends on strategy._precision_plugin to decide whether user passed None and we should choose precision plugin for them. 
This is what I'm trying to say in #11097, we want to distinguish what user passed in vs default value

Could you run tests/accelerator/test_accelerator_test.py? I think with this change, the custom plugin test will fail

Reply:
CI looks green

Reply:
I see why it worked. in accelerator_connector.py line 406
```
if (
            isinstance(self.strategy, TrainingTypePlugin) and self.strategy._precision_plugin is None
        ) or not isinstance(self.strategy, TrainingTypePlugin):
      call choose precision_plugin logic
```
Because training_type_plugin.precision_plugin haven't been called yet, the training_type_plugin._precision_plugin does not exist, so it guaranteed to be None, so it worked.

But this is VERY fragile, when the training_type_plugin.precision_plugin get called, both `training_type_plugin._precision_plugin` and `training_type_plugin.precision_plugin` will be the default value instead of what user set. If this happens before final_training_type_plugin(), even user themselves called it before pass into trainer, the precision plugin selection logic won't trigger and we will have silent bugs.  

As the test passes for now, feel free to land. We can discuss the longterm robust solution when we solve #11097

Reply:
I fully agree. This is not a long-term solution. The accelerator connector needs a revamp

Reply:
cc @awaelchli for this discussion, which is related to why ._accelerator become .accelerator